DigitalLearningSolutions.Data/Services/DelegateDownloadFileService.cs

```diff
@@ -74,7 +146,154 @@ private void PopulateJobGroupsSheet(IXLWorkbook workbook)
                     item => new { JobGroupID = item.id, JobGroupName = item.name }
                 );
 
-           ClosedXmlHelper.AddSheetToWorkbook(workbook, JobGroupsSheetName, jobGroups, TableTheme);
+            ClosedXmlHelper.AddSheetToWorkbook(workbook, JobGroupsSheetName, jobGroups, TableTheme);
+        }
+
+        private void PopulateAllDelegatesSheet(
+            IXLWorkbook workbook,
+            int centreId,
+            string? searchString,
+            string? sortBy,
+            string sortDirection,
+            string? filterString
+        )
+        {
+            var registrationPrompts = centreRegistrationPromptsService.GetCentreRegistrationPromptsByCentreId(centreId);
+            var delegatesToExport = GetDelegatesToExport(centreId, searchString, sortBy, sortDirection, filterString)
+                .ToList();
+
+            var dataTable = new DataTable();
+            // TODO: HEEDLS-810 - display extra PRN details if necessary. 
+            SetUpDataTableColumnsForAllDelegates(registrationPrompts, dataTable);
+
+            foreach (var delegateRecord in delegatesToExport)
+            {
+                SetDelegateRowValues(dataTable, delegateRecord, registrationPrompts);
+            }
+
+            if (dataTable.Rows.Count == 0)
+            {
+                var row = dataTable.NewRow();
+                dataTable.Rows.Add(row);
+            }
+
+            ClosedXmlHelper.AddSheetToWorkbook(
+                workbook,
+                AllDelegatesSheetName,
+                dataTable.AsEnumerable(),
+                XLTableTheme.None
+            );
+
+            FormatAllDelegateWorksheetColumns(workbook, dataTable);

```

Comment:
You could commonise this across all download files I think.
The format method can be commonised by adding a config param which contains the columns that need to be changed and their respective data types.

Reply:
Not easily. Only the Course Delegates download, and the new All Delegates download (which was based off the Course Delegates download) do things in exactly the same way. Every other place has its own subtlety by the looks of it, so again would probably be best suited for its own ticket I think.